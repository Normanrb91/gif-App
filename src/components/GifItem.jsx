export const GifItem = ({ gif }) => {
  return <div>{gif.title}</div>
}
