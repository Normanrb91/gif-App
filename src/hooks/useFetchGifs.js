import { useEffect, useState } from "react"
import { getGifs } from "../helper/getGifs"


export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading : true
    })

    useEffect(() => {
        getGifs(category)
        .then(gifs => {
            setState({
                data: gifs,
                loading: false
            })
        })
        .catch(() => {
            setState({
                ...data,
                loading: false
            })
        })

    }, [category])
    
    return state
}
