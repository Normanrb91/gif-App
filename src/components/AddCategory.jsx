import PropTypes from 'prop-types'
import { useForm } from '../hooks/useForm'

export const AddCategory = ({ onNewCategory }) => {
  const [values, handleInputChange, reset] = useForm({ category: '' })

  const handleSubmit = e => {
    e.preventDefault()
    if (values.category.trim().length <= 1) return

    onNewCategory(values.category)
    reset()
  }

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        placeholder="Buscar Gifs"
        name="category"
        type="text"
        value={values.category}
        onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
