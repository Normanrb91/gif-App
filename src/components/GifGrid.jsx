import { GifItem } from './GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>

      {loading ? (
        <h2>Cargando...</h2>
      ) : (
        <div className="card-grid">
          {data.map(gif => (
            <GifItem key={gif.id} {...gif} />
          ))}
        </div>
      )}
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
