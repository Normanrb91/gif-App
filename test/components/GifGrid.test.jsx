import { render, screen } from '@testing-library/react'
import { GifGrid } from '../../src/components/GifGrid'
import { useFetchGifs } from '../../src/hooks/useFetchGifs'

jest.mock('../../src/hooks/useFetchGifs')

describe('pruebas en <GifGrid />', () => {
  const category = 'Betis'

  test('debe de mostrar el loding inicial', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    render(<GifGrid category={category} />)
    expect(screen.getAllByText('Cargando...'))
    expect(screen.getAllByText(category))
  })

  test('debe de mostrar items cuando se cargan los gifs mediande el useGetGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        title: 'plantilla',
        url: 'https://betis.com/plantilla.jpg'
      },
      {
        id: '1123',
        title: 'Joaquin',
        url: 'https://taki.com/joquinfirma.jpg'
      }
    ]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })

    render(<GifGrid category={category} />)
    expect(screen.getAllByRole('img').length).toBe(2)
  })

  test('debe hacer match con snapshot', () => {
    const { container } = render(<GifGrid category={category} />)
    expect(container).toMatchSnapshot()
  })
})
