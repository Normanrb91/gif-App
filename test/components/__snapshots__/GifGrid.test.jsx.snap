// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe hacer match con snapshot 1`] = `
<div>
  <h3>
    Betis
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="plantilla"
        src="https://betis.com/plantilla.jpg"
      />
      <p>
        plantilla
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="Joaquin"
        src="https://taki.com/joquinfirma.jpg"
      />
      <p>
        Joaquin
      </p>
    </div>
  </div>
</div>
`;
